packages/@lwc/engine/dom/src/apis/build-custom-element-constructor.ts

```diff
@@ -38,3 +49,56 @@ export function deprecatedBuildCustomElementConstructor(
 
     return Ctor.CustomElementConstructor;
 }
+
+export function buildCustomElementConstructor(Ctor: ComponentConstructor): HTMLElementConstructor {
+    const def = getComponentInternalDef(Ctor);
+
+    // generating the hash table for attributes to avoid duplicate fields and facilitate validation
+    // and false positives in case of inheritance.
+    const attributeToPropMap: Record<string, string> = create(null);
+    for (const propName in def.props) {
+        attributeToPropMap[getAttrNameFromPropName(propName)] = propName;
+    }
+
+    return class extends def.bridge {
+        constructor() {
+            super();
+            createVM(this, def, {
+                mode: 'open',
+                isRoot: true,
+                owner: null,
+                tagName: this.tagName,
+            });
+        }
+        connectedCallback() {
+            connectRootElement(this);
+        }
+        disconnectedCallback() {
+            disconnectRootElement(this);
+        }
+        attributeChangedCallback(attrName: string, oldValue: string, newValue: string) {
+            if (oldValue === newValue) {
+                // Ignore same values.
+                return;
+            }
+            const propName = attributeToPropMap[attrName];
+            if (isUndefined(propName)) {
+                // Ignore unknown attributes.
+                return;
+            }
+            if (!isAttributeLocked(this, attrName)) {
+                // Ignore changes triggered by the engine itself during:
+                // * diffing when public props are attempting to reflect to the DOM
+                // * component via `this.setAttribute()`, should never update the prop.
```

Comment:
```suggestion
                // * component via `this.setAttribute()`, should never update the prop
```
removing the dot to be consistent w other comments